### PR TITLE
[CARBONDATA-3887] Fixed insert failure for global sort null data

### PR DESCRIPTION
**Pull request by kunal642**
### Why is this PR needed?
 Load data is failing with "Unsupported dataType: String" exception when null data is loaded into a string column in global_sort table as there is No handling for null data for string column
 
 ### What changes were proposed in this PR?
 Added a check for null data and handle for the same in global sort flow.

 ### Does this PR introduce any user interface change?
 - No
 - Yes. (please explain the change and update document)

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1561/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3298/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala [hunk 1]
```diff
@@ -921,52 +921,55 @@ object CommonUtil {
     var i = 0
     val fieldTypesLen = fields.length
     while (i < fieldTypesLen) {
-      if (!row.isNullAt(i)) {
-        fields(i).dataType match {
-          case StringType =>
-            data(i) = DataTypeUtil.getBytesDataDataTypeForNoDictionaryColumn(row.getString(i),
+      fields(i).dataType match {
```
Comment (akashrn5) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala:924]: I think initially, the if check of null at row was there for all datatypes, now handled for string, other data types will be handled?

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala [hunk 2]
```diff
@@ -468,6 +470,16 @@ class TestGlobalSortDataLoad extends QueryTest with BeforeAndAfterEach with Befo
       sql("SELECT * FROM carbon_localsort_difftypes ORDER BY shortField"))
   }
 
+  test("test global sort with null values") {
```
Comment (akashrn5) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala:473]: maybe can add a test case with other datatypes null value if not present in the current code.

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3341/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1601/

**Comment by brijoobopanna** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3349/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1609/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala (hunk 1 above)
Comment (kunal642) [reply integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala:924]: It is already handled for other types..Test cases is added

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala (hunk 2 above)
Comment (kunal642) [reply integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala:473]: done

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3366/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1625/

**Comment by ajantha-bhat** [comment]
LGTM

**Comment by akashrn5** [comment]
LGTM
